Tests/IntegrationTests/Infrastructure/PluginInspector.cs

```diff
@@ -0,0 +1,215 @@
+        {
+            if (logger == null)
+            {
+                throw new ArgumentNullException("logger");
+            }
+            if (string.IsNullOrWhiteSpace(tempDir))
+            {
+                throw new ArgumentNullException("tempDir");
+            }
+
+
+            this.logger = logger;
+            this.tempDir = tempDir;
+            this.jdkWrapper = new JdkWrapper();
+        }
+
+        public JarInfo GetPluginDescription(string jarFilePath)
+        {
+            Assert.IsTrue(File.Exists(jarFilePath), "Jar file does not exist");
+
+            this.Build(jarFilePath);
+
+            string reportFilePath = this.RunPluginInspector(jarFilePath, tempDir);
+
+            JarInfo jarInfo = null;
+            if (reportFilePath != null)
+            {
+                jarInfo = JarInfo.Load(reportFilePath);
+            }
+
+            return jarInfo;
+        }
+
+        private void CheckJdkIsInstalled()
+        {
+            if (!this.jdkWrapper.IsJdkInstalled())
+            {
+                Assert.Inconclusive("Test requires the JDK to be installed");
+            }
+        }
+
+        private void Build(string jarFilePath)
+        {
+            Assert.IsTrue(File.Exists(jarFilePath), "Jar file does not exist");
+
+            // Get the java source files
+            Directory.CreateDirectory(this.tempDir);
+            string srcDir = CreateSubDir(this.tempDir, "src");
+            string outDir = CreateSubDir(this.tempDir, "out");
+            string xxxDir = CreateSubDir(this.tempDir, "xxx");
+
+            SourceGenerator.CreateSourceFiles(this.GetType().Assembly,
+                "SonarQube.Plugins.IntegrationTests.Roslyn.Resources",
+                srcDir,
+                new Dictionary<string, string>());
+
+            JavaCompilationBuilder builder = new JavaCompilationBuilder(this.jdkWrapper);
+            foreach (string source in Directory.GetFiles(srcDir, "*.java", SearchOption.AllDirectories))
+            {
+                builder.AddSources(source);
+            }
+
+            // Add the jars required to compile the Java code
+            IEnumerable<string> jarFiles = GetCompileDependencies();
+
+            foreach (string jar in jarFiles)
+            {
+                builder.AddClassPath(jar);
+            }
+
+            bool result = builder.Compile(xxxDir, outDir, logger);
+
+            if (!result)
+            {
+                Assert.Inconclusive("Test setup error: failed to build the Java inspector");
+            }
+
+            this.inspectorClassFilePath = GetPluginInspectorClassFilePath(outDir);
+        }
+
+        private IEnumerable<string> GetCompileDependencies()
+        {
+            MavenPartialPOM pom = GetPOMFromResource(typeof(PluginInspector).Assembly, "SonarQube.Plugins.IntegrationTests.Roslyn.Resources.CompileTimeDependencies.pom");
+            IEnumerable<string> jarFiles = GetJarsFromPOM(pom);
+            return jarFiles;
```

Comment:
Currently, we embed the jar files required to build the plugin jar as resources in the plugin code. This isn't ideal, but it was acceptable since there were only three (although I think one of those isn't actually required)
However, actually executing the code in the generate jar requires far more jars to be present, and we don't want to embed them all.
Instead, we load a cut-down version of a POM file that specifies there required dependencies and fetch them from MavenCentral. Our currently POM is simplistic but sufficient for our current needs - all of the required dependencies must be specified explicitly. We don't currently support any of the following: variables, inheritance; dependency management, transitive dependencies, exclusions or conflict resolution.